### ISSUE TITLE
[MNG-4987] [regression] LATEST, RELEASE or SNAPSHOT version picked from wrong repository when resolution order does not match timestamp order

### ISSUE THREAD

**Issue by jira-importer** [apache/maven#7123]
**[Michael Hartmeier](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=mlhartme)** opened **[MNG-4987](https://issues.apache.org/jira/browse/MNG-4987?redirect=false)** and commented

In maven-aether-provider 3.0.2, DefaultVersionResolver, line 378ff says

```
private void merge( String key, Map<String, VersionInfo> infos, String timestamp, String version,
                    ArtifactRepository repository )
{
    VersionInfo info = infos.get( key );
    if ( info == null )
    {
        info = new VersionInfo( timestamp, version, repository );
        infos.put( key, info );
    }
    else if ( info.isOutdated( timestamp ) )
    {
        info.version = version;
        info.repository = repository;
    }
}
```

If I understand correctly, you should add

```
info.timestamp = timestamp
```

to the else part. Otherwise, you'll use a wrong timestamp in future calls to this method.

---

**Affects:** 3.0.2

**Comment by jira-importer** [comment]
**[Michael Hartmeier](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=mlhartme)** commented

It probably not a major bug, but I don't see where I can change that now

**Comment by jira-importer** [comment]
**[Benjamin Bentmann](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=bentmann)** commented

Good catch, fixed in [1073807](http://svn.apache.org/viewvc?view=revision&revision=1073807), and given SNAPSHOT resolution is also affected, we can safely call this a major bug.
